force-app/main/default/classes/ITriggerHandler.cls

```diff
@@ -0,0 +1,31 @@
+public interface ITriggerHandler {
+	void beforeUpdate(
+		Map<Id, SObject> newItemsMap,
+		Map<Id, SObject> oldItemsMap,
+		List<SObject> newItems,
+		List<SObject> oldItems
+	);
+
+	void beforeInsert(List<SObject> newItems);
+
+	void beforeDelete(Map<Id, SObject> oldItemsMap);
+
+	void afterUpdate(
+		Map<Id, SObject> newItemsMap,
+		Map<Id, SObject> oldItemsMap,
+		List<SObject> newItems,
+		List<SObject> oldItems
+	);
+
+	void afterDelete(Map<Id, SObject> oldItemsMap);
+
+	void afterUndelete(Map<Id, SObject> oldItemsMap);
+
+	void afterInsert(Map<Id, SObject> newItemsMap);
+
+	Boolean isTriggerEnabled();
+
+	void disableTrigger();
```

Comment:
we can use abstract class instead of implementing disableTrigger() enableTrigger() in every next class that implements this interface. As implementation of these methods is default for all next TriggerHandlers